R2API/ItemAPI.cs

```diff
@@ -183,6 +187,29 @@ public static class ItemAPI {
             return false;
         }
 
+        /// <summary>
+        /// Add a custom item tag to the list of item tags.
+        /// If this is called after the ItemCatalog inits then this will return -1 and ignore the custom item tag.
+        /// </summary>
+        /// <param name="name">The tag to add.</param>
+        /// <returns>ItemTag value if added or already existent, (-1) cast to ItemTag otherwise</returns>
+        public static ItemTag AddItemTag(string name){
+            if (!Loaded) {
+                throw new InvalidOperationException($"{nameof(ItemAPI)} is not loaded. Please use [{nameof(R2APISubmoduleDependency)}(nameof({nameof(ItemAPI)})]");
+            }
+
+            if (!CatalogBlockers.GetAvailability<ItemDef>()) {
+                R2API.Logger.LogError($"Too late ! Tried to add itemTag: {name} after the ItemCatalog has Initialized!");
+                return (ItemTag)(-1);
+            }
+            var result = customItemTags.IndexOf(name) + 1;
```

Comment:
Should use `FindItemTagByName` here and then add to `customItemTags` if result is -1, because otherwise you are able to add custom tag with the same name as vanilla, but then you will not be able to retrieve it from the API which will lead to a confusion

Reply:
Actually, no, the other way around, you will be able to get custom tag from `FindItemTagByName`, but not vanilla

Reply:
Good Point,thanks.